src/advanced_inv.cpp

```diff
@@ -1966,3 +2020,77 @@ void advanced_inventory::draw_minimap()
         g->u.draw(minimap, g->u.posx(), g->u.posy(), player_invert);
     }
 }
+
+void advanced_inventory::set_vehicle(aim_location sel)
+{
+    // setup on the square itself
+    squares[AIM_VEHICLE].set_vehicle(squares[sel]);
+    // setup for post AIM_VEHICLE selection
+    panes[src].area   = AIM_VEHICLE;
+    panes[src].index  = 0;
+    panes[src].recalc = true;
+    panes[src].redraw = true;
+}
+
+void advanced_inv_area::set_vehicle(advanced_inv_area &s)
+{
+    // this only works for the AIM_VEHICLE square!
+    if(id != AIM_VEHICLE) {
+        return;
+    }
+    if(s.veh == nullptr) {
+        debugmsg("[bug]: squares[sel].veh == nullptr!");
+        return;
```

Comment:
OK, I just got this when having a veh to my NW/7 position, with the V slot Red'd out and me hitting it anyway.

(Debug messages shouldn't show up for routinely-not-suitable uses: something like "This is only available on a vehicle tile" as a conventional Nope message would be suitable.  I may change this in the merge, depending on the other PRs checking out, but feel free to do so and post so I know to re-pull.)